### PR TITLE
raise error when hist on non-numeric cols

### PR DESCRIPTION
**Pull request by dvgodoy**
This PR fixes bug #502 , raising the proper TypeError when `hist()` is called on a non-numeric column (just like pandas does).
I also had to fix another (AFAIK unreported) bug in `select_dtypes` in the process, as it was not properly appending columns to the list.

### CODE REVIEW

File: databricks/koalas/frame.py [hunk 1]
```diff
@@ -3245,7 +3245,7 @@ def select_dtypes(self, include=None, exclude=None):
                     self._sdf.schema[col].dataType in exclude_spark_type)
 
             if should_include:
-                columns += col
+                columns.append(col)
```
Comment (HyukjinKwon) [inline databricks/koalas/frame.py]: Oh, nice catch.

File: databricks/koalas/plot.py [hunk 2]
```diff
@@ -376,6 +378,25 @@ def calc_histogram(self, bins):
 
 class KoalasHistPlot(HistPlot):
     def _args_adjust(self):
+        from databricks.koalas.series import Series
+
+        data = self.data
+        if isinstance(data, Series):
+            data = data.to_frame()
+
+        numeric_data = data.select_dtypes(include=['byte', 'decimal', 'integer', 'float',
```
Comment (HyukjinKwon) [inline databricks/koalas/plot.py]: I think we can simply check `dtypes`.

File: databricks/koalas/plot.py [hunk 3]
```diff
@@ -376,6 +378,25 @@ def calc_histogram(self, bins):
...
+
+        try:
+            is_empty = numeric_data.empty
+        except AttributeError:
+            is_empty = not len(numeric_data)
+
+        # no empty frames or series allowed
+        if is_empty:
+            raise TypeError('Empty {0!r}: no numeric data to '
+                            'plot'.format(numeric_data.__class__.__name__))
```
Comment (HyukjinKwon) [inline databricks/koalas/plot.py]: Can we add a simple test to prevent the mistake like the one in select_dtypes?

**Comment by dvgodoy** [comment]
@HyukjinKwon 
I've just added tests to it - sorry for taking so long, but I went on holidays :-)

**Comment by HyukjinKwon** [comment]
Sorry for late input. I rebased and opened a PR at https://github.com/databricks/koalas/pull/766.
I will credit to you.

**Pull request closed without merging**